erpnext/accounts/doctype/fixed_asset_account/depreciation_report.py

```diff
@@ -0,0 +1,154 @@
+from erpnext.accounts.utils import get_fiscal_year
+from frappe.utils import flt
+
+def getDateDiffDays(d1, d2):
+    return abs((d2 - d1).days)
+    
+def getDeprTillLastYear(day_before_start, fixed_asset_name):
+        for accdepr in frappe.get_doc("Fixed Asset Account", fixed_asset_name).depreciation:
+           if get_fiscal_year(fiscal_year = accdepr.fiscal_year) == get_fiscal_year(date = day_before_start):
+               return accdepr.total_accumulated_depreciation
+        return 0
+        
+def getDeprProvidedThisYear(method, depronopening, purchase_value, deprtilllastyr, rateofdepr):
+	if method == "Written Down" or (depronopening <= (purchase_value - deprtilllastyr)):
+		return flt(depronopening,2)
+	elif depronopening > (purchase_value - deprtilllastyr) and \
+	(purchase_value - deprtilllastyr) < (purchase_value * rateofdepr / 100) and \
+	method=="Straight Line":
+		return flt(purchase_value - deprtilllastyr,2)
+	elif method=="Straight Line":
+		return 0
+
+def get_report_data(finyrfrom, finyrto, company, fa_name=None):
+    data = []
+    finyrfrom = datetime.strptime(finyrfrom, "%Y-%m-%d").date()
+    finyrto = datetime.strptime(finyrto, "%Y-%m-%d").date()
+    day_before_start = finyrfrom - timedelta (days=1)    
+    TOTAL_DAYS_IN_YEAR = 365
+    method = frappe.get_doc("Company", company).default_depreciation_method
+    global ps
+    if fa_name==None:
+	    ps = frappe.db.sql("""select led.*,ifnull((select sum(sale.asset_purchase_cost) from `tabFixed Asset Sale` sale where sale.fixed_asset_account=led.fixed_asset_name and sale.docstatus=1  and sale.posting_date>=%s and sale.posting_date<=%s),0) as total_sale_value
+           from `tabFixed Asset Account` led where is_sold=false or (is_sold=true and (select count(*) from `tabFixed Asset Sale` sale where sale.fixed_asset_account=led.fixed_asset_name and docstatus=1 and sale.posting_date>=%s and sale.posting_date<=%s)>0)""", (finyrfrom,finyrto,finyrfrom,finyrto), as_dict=True)
+    else:
+	    ps = frappe.db.sql("""select led.*,ifnull((select sum(sale.asset_purchase_cost) from `tabFixed Asset Sale` sale where sale.fixed_asset_account=led.fixed_asset_name and sale.docstatus=1  and sale.posting_date>=%s and sale.posting_date<=%s),0) as total_sale_value
+           from `tabFixed Asset Account` led where (is_sold=false or (is_sold=true and (select count(*) from `tabFixed Asset Sale` sale where sale.fixed_asset_account=led.fixed_asset_name and docstatus=1 and sale.posting_date>=%s and sale.posting_date<=%s)>0)) and led.fixed_asset_name=%s""", (finyrfrom,finyrto,finyrfrom,finyrto,fa_name), as_dict=True)
+
+
+    for assets in ps:
+        fixed_asset_name = assets.fixed_asset_name
+        fixed_asset_account = assets.fixed_asset_account
+        rateofdepr = abs(assets.depreciation_rate)
+        purchase_date = datetime.strptime(assets.purchase_date, "%Y-%m-%d").date()
+
+        global deprtilllastyr # Depreciation provided till close of last fiscal Yr.

```

Comment:
very bad design to use globals, wrap it in a class if you want to share variables